From 77bb0b4aaa08a9ec64fc6289c8d661c1f8af5736 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:35:53 -0400
Subject: [block] blk-mq: select random tag betweet 0 and (depth - 1)

Message-id: <1402670238-13916-66-git-send-email-snitzer@redhat.com>
Patchwork-id: 83846
O-Subject: [RHEL7.1 PATCH 065/150] blk-mq: bitmap tag: select random tag betweet 0 and (depth - 1)
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 9d3d21aeb4f194cd7ac205abe68b14b47ae736a8
Author: Ming Lei <tom.leiming@gmail.com>
Date:   Sat May 10 15:43:14 2014 -0600

    blk-mq: bitmap tag: select random tag betweet 0 and (depth - 1)

    The selected tag should be selected at random between 0 and
    (depth - 1) with probability 1/depth, instead between 0 and
    (depth - 2) with probability 1/(depth - 1).

    Signed-off-by: Ming Lei <tom.leiming@gmail.com>
    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq-tag.c b/block/blk-mq-tag.c
index 5a83d8e..f196e60 100644
--- a/block/blk-mq-tag.c
+++ b/block/blk-mq-tag.c
@@ -449,10 +449,7 @@ void blk_mq_tag_init_last_tag(struct blk_mq_tags *tags, unsigned int *tag)
 {
  unsigned int depth = tags->nr_tags - tags->nr_reserved_tags;
 
- if (depth > 1)
-  *tag = prandom_u32() % (depth - 1);
- else
-  *tag = 0;
+ *tag = prandom_u32() % depth;
 }
 
 ssize_t blk_mq_tag_sysfs_show(struct blk_mq_tags *tags, char *page)
-- 
1.7.1